### PR TITLE
fix: speed improvement

### PR DESCRIPTION
**Pull request by zekth**

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: index.js [hunk 1]
```diff
@@ -113,28 +113,13 @@ function serialize (cmpts, opts) {
...
   const schemeHandler = SCHEMES[(options.scheme || components.scheme || '').toLowerCase()]
 
   // perform scheme specific serialization
   if (schemeHandler && schemeHandler.serialize) schemeHandler.serialize(components, options)
 
-  if (components.host) {
-    // if host component is an IPv6 address
-    // if (protocol.IPV6ADDRESS.test(components.host)) {
-    // TODO: normalize IPv6 address as per RFC 5952
-    // }
```
Comment (jsumners) [inline index.js:126]: I've looked at tackling this issue (writing a better URI parser), and this is where the difficulty lies. The problems are in sections:

+ https://www.rfc-editor.org/rfc/rfc5952#section-2.1
+ https://www.rfc-editor.org/rfc/rfc5952#section-2.2

(Technically, an IPv4 address can _also_ be compressed, but is rarely done.)

The result is that most implementations inspect the string format determine if the string is written correctly according to the various possible string formats. Which results in the regex found in the original library, along with things like:

+ https://stackoverflow.com/questions/53497/regular-expression-that-matches-valid-ipv6-addresses

We can also see parsing implementations in:

+ https://gitbox.apache.org/repos/asf?p=commons-validator.git;a=blob;f=src/main/java/org/apache/commons/validator/routines/InetAddressValidator.java;h=357386d617efcb6724143aa3743bbe045acdedf7;hb=HEAD#l130
+ https://github.com/golang/go/blob/9e6ad46bccfa7a63e768236bcd1fd54dab38e4d1/src/net/ip.go#L597-L697

Comment (zekth) [reply index.js:126]: This code is also dead in the original lib. Also the parsing of the IPV6 is done here: https://github.com/fastify/fast-uri/blob/5da3a5127b916c2c159893156220e48f4fb3a169/lib/utils.js#L21-L85
Which works for all the current test from the original lib. I feel there might be some edge cases we did not find yet.

File: README.md [hunk 2]
```diff
@@ -5,26 +5,25 @@ Dependency free RFC 3986 URI toolbox.
...
+urijs: parse IPv4 x 387,909 ops/sec ±0.13% (97 runs sampled)
+fastURI: parse IPv6 x 884,259 ops/sec ±0.36% (99 runs sampled)
+urijs: parse IPv6 x 289,415 ops/sec ±0.12% (100 runs sampled)
+fastURI: serialize uri x 1,784,977 ops/sec ±0.15% (98 runs sampled)
+urijs: serialize uri x 392,667 ops/sec ±0.37% (96 runs sampled)
+fastURI: serialize IPv6 x 452,999 ops/sec ±0.15% (101 runs sampled)
+urijs: serialize IPv6 x 259,007 ops/sec ±0.59% (93 runs sampled)
+fastURI: serialize ws x 1,439,760 ops/sec ±0.10% (99 runs sampled)
+urijs: serialize ws x 346,635 ops/sec ±0.16% (97 runs sampled)
+fast-uri: resolve x 342,474 ops/sec ±0.97% (99 runs sampled)
```
Comment (Eomm) [inline README.md:20]: Should not it be `fastURI`?

Comment (zekth) [reply README.md:20]: my bad. Will fix.
